coderd/coderd.go

```diff
@@ -287,6 +291,11 @@ func New(options *Options) *API {
 		v := schedule.NewAGPLUserQuietHoursScheduleStore()
 		options.UserQuietHoursScheduleStore.Store(&v)
 	}
+	ctx, cancel := context.WithCancel(context.Background())
+
+	if options.StatsBatcher == nil {
+		panic("developer error: options.StatsBatcher is nil")
```

Comment:
Is there a reason why `New(options)` can't define a default instance?

Reply:
No, just keeping to the existing pattern in this file.